### PR TITLE
API to web interface (draft)

### PR DESCRIPTION
**Pull request by Marge-m**

### CODE REVIEW

File: backend/settings.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+from os.path import isfile
```
Comment (mvartanyan) [inline backend/settings.py]: better not import names directly from stdlib

File: backend/settings.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+from envparse import env
```
Comment (mvartanyan) [inline backend/settings.py]: wrong import order

File: backend/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+aiohttp==3.5.4
```
Comment (mvartanyan) [inline backend/requirements.txt]: can we have one list of requirements for everything?

File: backend/auth/views.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import json
+from time import time
+from bson.objectid import ObjectId
```
Comment (mvartanyan) [inline backend/auth/views.py]: where is this from / what is this for?

File: backend/auth/models.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# from bson.objectid import ObjectId
```
Comment (mvartanyan) [inline backend/auth/models.py]: remove dead code

File: backend/auth/views.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+from aiohttp import web
+from aiohttp_session import get_session
+from aiohttp_security import remember, forget, authorized_userid
+
+
+def redirect(request, router_name):
```
Comment (mvartanyan) [inline backend/auth/views.py]: I think we have agreed to use type annotations?

File: backend/auth/views.py [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    url = request.app.router[router_name].url()
+    raise web.HTTPFound(url)
+
+
+def set_session(session, username, request):
```
Comment (mvartanyan) [inline backend/auth/views.py]: a little docstring would be good in this case. I think that I understand what the method does (from reading the code, not from the method name). And I suspect that the choice of method name is not great. But I would prefer not to pass judgment before I see the author's intent :-)

File: backend/auth/views.py [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+        return 'username is required'
+    return None
+
+
+class Login(web.View):
+    async def get(self):
+        session = await get_session(self.request)
+        if session.get('user'):
+            redirect(self.request, 'main')
+        return {'content': 'Please enter login or email'}
```
Comment (mvartanyan) [inline backend/auth/views.py:40]: Let's not have string literals scattered across code

File: backend/auth/views.py [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+        session = await get_session(self.request)
+        set_session(session, data['username'], self.request)
+
+
+class Logout(web.View):
+    async def logout(request):
+        response = redirect(request.app.router, 'login')
+        await forget(request, response)
+        return response
```
Comment (mvartanyan) [inline backend/auth/views.py]: flake8 it!

**Comment by mvartanyan** [comment]
Please merge in the master branch so CI can work.

File: backend/app.py [hunk 10]
```diff
@@ -0,0 +1,13 @@
+#! /usr/bin/env python3.7
```
Comment (mjecov) [inline backend/app.py]: Not sure that we really need this

File: backend/auth/views.py (hunk 9 above)
Comment (mjecov) [inline backend/auth/views.py]: add new line at the end of the file please!

File: backend/routes.py [hunk 11]
```diff
@@ -0,0 +1,6 @@
+from auth.views import Login, Logout
+
+routes = [
+    ('*',   '/login',   Login,     'login'),
+    ('*',   '/logout', Logout,   'logout'),
+]
```
Comment (mjecov) [inline backend/routes.py]: add blank line please

**Review by mjecov — CHANGES_REQUESTED** [review]

**Comment by apatrushev** [comment]
Are this still relevant?

**Pull request closed without merging**